src/DGtal/geometry/curves/ArithmeticalDSS.h

```diff
@@ -497,14 +502,49 @@ namespace DGtal
      * @param aPoint any point
      */
     bool isInDSS(const Point& aPoint) const; 
-
-    /**
-     * @return 'true' if @a aPoint is in the DSS
-     * 'false' otherwise. 
-     * @param aPoint any point
-     * @see isInDSS
-     */
+    
     bool operator()(const Point& aPoint) const; 
+    
+    
+    /** 
+     * Returns a boolean equal to true if 'this' belongs to the DSL @a aDSL, false otherwise.
+     * @return 'true' if 'this' belongs to the DSL @a aDSL.
+     * 'false' otherwise.
+     * @param aDSL any DSL
+     */
+    bool isInDSL(const DSL& aDSL) const;
+
+ 
+    /** 
+     * Returns a boolean equal to true if 'this' belongs to the DSL @a
+     * aDSL, false otherwise. Also returns extra information about the
+     * leaning points included in 'this' or a point outside @a aDSL. 
+     * @return 'true' if 'this' belongs to the DSL @a aDSL.
+     * 'false' otherwise.
+     * @param aDSL any DSL
+     * @return Ulp the list of @a aDSL upper leaning points on 'this', if any
+     * @return Llp the list of @a aDSL lower leaning points on 'this', if any
+     * @return outP a point of 'this' that does not belong to @a aDSL, if any
+     */
```

Comment:
Ideally, there is a singe @return corresponding to the function return. For input/output parameters, you can use:

     @param [in] aDSL  ....

or  

     @param [in,out] aDSL....

or 

     @param [out] aDSL;..